org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -174,10 +173,10 @@ public boolean visit(IResourceProxy proxy) throws CoreException {
       }
       for (final IFile file : filesToAnalyze) {
         final java.nio.file.Path filePath = file.getRawLocation().makeAbsolute().toFile().toPath();
-        inputFiles.add(new InputFile() {
+        inputFiles.add(new ClientInputFile() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)